sklearn/metrics/_plot/tests/test_common_curve_display.py

```diff
@@ -86,7 +86,7 @@ def fit(self, X, y):
 
     clf = MyClassifier().fit(X, y)
 
-    with pytest.raises(ValueError, match=msg):
+    with pytest.raises(AttributeError, match=msg):
```

Comment:
This is not backward compatible but I think it's fine to change the type of error